linera-chain/src/certificate/confirmed.rs

```diff
@@ -13,19 +13,19 @@ use serde::{ser::SerializeStruct, Deserialize, Deserializer, Serialize};
 
 use super::{generic::GenericCertificate, Certificate};
 use crate::{
-    block::{ConfirmedBlock, ConversionError},
-    data_types::{ExecutedBlock, Medium, MessageBundle},
+    block::{Block, ConfirmedBlock, ConversionError},
+    data_types::{Medium, MessageBundle},
 };
 
 impl GenericCertificate<ConfirmedBlock> {
     /// Returns reference to the `ExecutedBlock` contained in this certificate.
-    pub fn executed_block(&self) -> &ExecutedBlock {
-        self.inner().executed_block()
+    pub fn block(&self) -> &Block {
+        self.inner().block()
     }
 
     /// Returns whether this value contains the message with the specified ID.
     pub fn has_message(&self, message_id: &MessageId) -> bool {
-        self.executed_block().message_by_id(message_id).is_some()
+        self.block().message_by_id(message_id).is_some()
```

Comment:
Would it make sense the handle the renaming `executed_block` -> `block` first in a separate PR?

Reply:
Actually, I didn't get rid of `ExecutedBlock`. I tried but the `Proposal`/`BlockExecutionOutcome` is used in a couple of places (we execute before proposing, we execute when we receive a proposal, we re-execute when we receive a confirmed certificate, etc.) so I kept the `ExecutedBlock` as an internal data structure that caries the result of execution.